webapp/channels/src/reducers/plugins/index.ts

```diff
@@ -385,6 +386,41 @@ function siteStatsHandlers(state: PluginsState['siteStatsHandlers'] = {}, action
     }
 }
 
+function userSettings(state: PluginsState['userSettings'] = {}, action: GenericAction) {
+    switch (action.type) {
+    case ActionTypes.RECEIVED_PLUGIN_USER_SETTINGS:
+        if (action.data) {
+            const extractedConfiguration = extractPluginConfiguration(action.data.setting);
+            if (!extractedConfiguration) {
+                // eslint-disable-next-line no-console
+                console.warn(`Plugin ${action.data.pluginId} is trying to register an invalid configuration. Contact the plugin developer to fix this issue.`);
+                return state;
+            }
+            if (extractedConfiguration.id !== action.data.pluginId) {
```

Comment:
Do we need to duplicate the ID in the configuration that the plugin has to pass? Perhaps we could leave that field out and have `extractPluginConfiguration` add it instead, or we could even remove the ID from that object and pass it to any components that need it separately

Reply:
I like it this way for simplicity, but 0/5. Also, we could make the plugin not define the id at all, or always overwrite it at this level (similar with the UIName, that we can get from the plugin manifest).

But really, no strong opinions here.